File: testutils/src/main/java/com/mparticle/mock/MockApplication.java
package com.mparticle.mock;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.content.res.Resources;

import java.io.File;

/**
 * Created by sdozor on 4/10/15.
 */
public class MockApplication extends Application {
    MockContext mContext;
    public ActivityLifecycleCallbacks mCallbacks;

    public MockApplication(MockContext context) {
        super();
        mContext = context;
    }

    @Override
    public void registerActivityLifecycleCallbacks(ActivityLifecycleCallbacks callback) {
        mCallbacks = callback;
    }

    @Override
    public Context getApplicationContext() {
        return this;
    }

    public void setSharedPreferences(SharedPreferences prefs) {
        mContext.setSharedPreferences(prefs);
    }

    @Override
    public Object getSystemService(String name) {
        return mContext.getSystemService(name);
    }

    @Override
    public SharedPreferences getSharedPreferences(String name, int mode) {
        return mContext.getSharedPreferences(name, mode);
    }

    @Override
    public PackageManager getPackageManager() {
        return mContext.getPackageManager();
    }

    @Override
    public String getPackageName() {
        return mContext.getPackageName();
    }

    @Override
    public ApplicationInfo getApplicationInfo() {
        return mContext.getApplicationInfo();
    }

    @Override
    public Resources getResources() {
        return mContext.getResources();
    }

    @Override
    public File getFilesDir() {
        return mContext.getFilesDir();
    }
}

File: kits/rokt/rokt/src/test/kotlin/com/mparticle/kits/mocks/MockApplication.kt
package com.mparticle.kits.mocks

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.content.res.Resources

class MockApplication(var mContext: MockContext) : Application() {
    var mCallbacks: ActivityLifecycleCallbacks? = null
    override fun registerActivityLifecycleCallbacks(callback: ActivityLifecycleCallbacks) {
        mCallbacks = callback
    }

    override fun getApplicationContext(): Context = this

    fun setSharedPreferences(prefs: SharedPreferences) {
        mContext.setSharedPreferences(prefs)
    }

    override fun getSystemService(name: String): Any? = mContext.getSystemService(name)

    override fun getSharedPreferences(name: String, mode: Int): SharedPreferences =
        mContext.getSharedPreferences(name, mode)

    override fun getPackageManager(): PackageManager = mContext.packageManager

    override fun getPackageName(): String = mContext.packageName

    override fun getApplicationInfo(): ApplicationInfo = mContext.applicationInfo

    override fun getResources(): Resources = mContext.resources
}
